neuraxle/checkpoints.py

```diff
@@ -596,9 +598,302 @@ def __init__(self):
+        """
+        Read a data input checkpoint with :class:`BaseDataCheckpointer`.
+
+        :param context: execution context
+        :type context: ExecutionContext
+        :param current_id: current id
+        :type current_id: str
+        :param data_input: data input
+        :type data_input: Any
+        :param expected_output: expected output
+        :type expected_output: Any
+        :return: current_id, data_input, expected_output
+        :rtype: Tuple[str, Any, Any]
+        """
+        checkpoint = self.data_checkpointer.read_checkpoint(
+            checkpoint_path=self._get_checkpoint_path(context),
+            current_id=current_id
+        )
+
+        return current_id, checkpoint, expected_output
+
+
+class ExpectedOutputCheckpointerWrapper(BaseDataCheckpointerWrapper):
+    """
+    Checkpoint expected output with data checkpointer.
+
+    .. seealso::
+        * :class:`BaseDataCheckpointerWrapper`
+    """
+
+    def __init__(self, data_checkpointer: BaseDataCheckpointer):
+        BaseDataCheckpointerWrapper.__init__(
+            self,
+            data_checkpointer=data_checkpointer,
+            checkpoint_type_name=DataCheckpointType.EXPECTED_OUTPUT.value
+        )
+
+    def save_one_checkpoint(self, context: ExecutionContext, current_id: str, data_input: Any, expected_output: Any):
+        """
+        Save one expected output data checkpoint with :class:`BaseDataCheckpointer`.
+
+        :param context: execution context
+        :type context: ExecutionContext
+        :param current_id: current id
+        :type current_id: str
+        :param data_input: data input
+        :type data_input: Any
+        :param expected_output: expected output
+        :type expected_output: Any
+        :return:
+        """
+        self.data_checkpointer.save_checkpoint(
+            checkpoint_path=self._get_checkpoint_path(context),
+            current_id=current_id,
+            data=expected_output
+        )
+
+    def read_one_checkpoint(self, context: ExecutionContext, current_id: str, data_input: Any, expected_output: Any) -> Tuple:
+        """
+        Read an expected output checkpoint with a :class:`BaseDataCheckpointer`.
+
+        :param context: execution context
+        :type context: ExecutionContext
+        :param current_id: current id
+        :type current_id: str
+        :param data_input: data input
+        :type data_input: Any
+        :param expected_output: expected output
+        :type expected_output: Any
+        :return: current_id, data_input, expected_output
+        :rtype: Tuple[str, Any, Any]
+        """
+        checkpoint = self.data_checkpointer.read_checkpoint(
+            checkpoint_path=self._get_checkpoint_path(context),
+            current_id=current_id
+        )
+
+        return current_id, data_input, checkpoint
+
+class FullCheckpoint(Pipeline):
```

Comment:
@guillaume-chevalier I think we should name this to something like ExecutionCheckpoint or ExecutionPickleCheckpoint ? because the checkpoint is different for each execution mode ? maybe rename the other one to VanillaCheckpoint or NormalCheckpoint dunno just an idea

Reply:
> "maybe rename the other one to VanillaCheckpoint or NormalCheckpoint dunno just an idea"

What is the `other` you refer to here?

What about `FullExecutionCheckpoint` ? because just saying `execution` is not clear that it's in every execution case.